FAILED: ReliableProxyMultiJvm

```
[JVM-1] - must stop proxy if target stops and no reconnection *** FAILED ***
[JVM-1]   akka.actor.InvalidActorNameException: actor name [proxy] is not unique!
[JVM-1]   at akka.actor.dungeon.ChildrenContainer$TerminatingChildrenContainer.reserve(ChildrenContainer.scala:192)
[JVM-1]   at akka.actor.dungeon.Children$class.reserveChild(Children.scala:76)
[JVM-1]   at akka.actor.ActorCell.reserveChild(ActorCell.scala:369)
[JVM-1]   at akka.actor.dungeon.Children$class.makeChild(Children.scala:202)
[JVM-1]   at akka.actor.dungeon.Children$class.attachChild(Children.scala:41)
[JVM-1]   at akka.actor.ActorCell.attachChild(ActorCell.scala:369)
[JVM-1]   at akka.actor.ActorSystemImpl.actorOf(ActorSystem.scala:586)
[JVM-1]   at akka.contrib.pattern.ReliableProxySpec$$anonfun$1$$anonfun$apply$mcV$sp$38$$anonfun$apply$mcV$sp$39.apply$mcV$sp(ReliableProxySpec.scala:292)
[JVM-1]   at akka.remote.testkit.MultiNodeSpec.runOn(MultiNodeSpec.scala:338)
[JVM-1]   at akka.contrib.pattern.ReliableProxySpec$$anonfun$1$$anonfun$apply$mcV$sp$38.apply$mcV$sp(ReliableProxySpec.scala:288)
[JVM-1]   ...
[JVM-1] [ERROR] [02/26/2015 08:36:18.495] [ReliableProxySpec-akka.actor.default-dispatcher-3] [akka://ReliableProxySpec/user/controller/barriers] RoleName(local) tried to enter 'after-each' while we were waiting for 'test7a' (akka.remote.testconductor.BarrierCoordinator$WrongBarrier)
[JVM-1] [WARN] [02/26/2015 08:36:18.501] [ReliableProxySpec-akka.actor.default-dispatcher-14] [akka://ReliableProxySpec/user/controller/barriers] cannot remove RoleName(local): no client to remove
[JVM-2] - must stop proxy if target stops and no reconnection *** FAILED ***
[JVM-2]   java.lang.RuntimeException: barrier failed: test7a
[JVM-2]   at akka.remote.testconductor.ClientFSM$$anonfun$3.applyOrElse(Player.scala:211)
[JVM-2]   at akka.remote.testconductor.ClientFSM$$anonfun$3.applyOrElse(Player.scala:186)
[JVM-2]   at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
[JVM-2]   at akka.actor.FSM$class.processEvent(FSM.scala:621)
[JVM-2]   at akka.remote.testconductor.ClientFSM.akka$actor$LoggingFSM$$super$processEvent(Player.scala:147)
[JVM-2]   at akka.actor.LoggingFSM$class.processEvent(FSM.scala:752)
[JVM-2]   at akka.remote.testconductor.ClientFSM.processEvent(Player.scala:147)
[JVM-2]   at akka.actor.FSM$class.akka$actor$FSM$$processMsg(FSM.scala:615)
[JVM-2]   at akka.actor.FSM$$anonfun$receive$1.applyOrElse(FSM.scala:609)
[JVM-2]   at akka.actor.Actor$class.aroundReceive(Actor.scala:465)
```

https://jenkins.akka.io:8498/job/akka-multi-node-nightly/2109

--- a/akka-contrib/src/multi-jvm/scala/akka/contrib/pattern/ReliableProxySpec.scala
+++ b/akka-contrib/src/multi-jvm/scala/akka/contrib/pattern/ReliableProxySpec.scala
@@ -59,9 +59,10 @@ class ReliableProxySpec extends MultiNodeSpec(ReliableProxySpec) with STMultiNod
   }
 
   def stopProxy(): Unit = {
-    proxy ! FSM.UnsubscribeTransitionCallBack(testActor)
-    system stop proxy
-    expectMsgType[Terminated]
+    val currentProxy = proxy
+    currentProxy ! FSM.UnsubscribeTransitionCallBack(testActor)
+    currentProxy ! PoisonPill
+    expectTerminated(currentProxy)
   }
 
   def expectState(s: State) = expectMsg(FSM.CurrentState(proxy, s))
@@ -91,7 +92,7 @@ class ReliableProxySpec extends MultiNodeSpec(ReliableProxySpec) with STMultiNod
         import akka.contrib.pattern.ReliableProxy
 
         idTarget()
-        proxy = system.actorOf(ReliableProxy.props(target.path, 100.millis, 5.seconds), "proxy")
+        proxy = system.actorOf(ReliableProxy.props(target.path, 100.millis, 5.seconds), "proxy1")
         watch(proxy)
         proxy ! FSM.SubscribeTransitionCallBack(testActor)
         expectState(Connecting)
@@ -289,7 +290,7 @@ class ReliableProxySpec extends MultiNodeSpec(ReliableProxySpec) with STMultiNod
         stopProxy() // Stop previous proxy
 
         // Start new proxy with no reconnections
-        proxy = system.actorOf(ReliableProxy.props(target.path, 100.millis), "proxy")
+        proxy = system.actorOf(ReliableProxy.props(target.path, 100.millis), "proxy2")
         proxy ! FSM.SubscribeTransitionCallBack(testActor)
         watch(proxy)
 
@@ -308,7 +309,7 @@ class ReliableProxySpec extends MultiNodeSpec(ReliableProxySpec) with STMultiNod
       runOn(local) {
         within(5 seconds) {
           expectMsgType[ProxyTerminated]
-          expectMsgType[Terminated]
+          expectTerminated(proxy)
         }
       }
 
@@ -332,7 +333,7 @@ class ReliableProxySpec extends MultiNodeSpec(ReliableProxySpec) with STMultiNod
       runOn(local) {
         // Proxy is not running after previous test
         // Start new proxy with 3 reconnections every 2 sec
-        proxy = system.actorOf(ReliableProxy.props(target.path, 100.millis, 2.seconds, 3), "proxy")
+        proxy = system.actorOf(ReliableProxy.props(target.path, 100.millis, 2.seconds, 3), "proxy3")
         proxy ! FSM.SubscribeTransitionCallBack(testActor)
         watch(proxy)
         expectState(Connecting)
@@ -364,7 +365,7 @@ class ReliableProxySpec extends MultiNodeSpec(ReliableProxySpec) with STMultiNod
           // Validate that the unsent messages are 50 ints
           val unsentInts = proxyTerm.outstanding.queue collect { case Message(i: Int, _, _) if i > 0 && i <= 50 ⇒ i }
           unsentInts should have size 50
-          expectMsgType[Terminated]
+          expectTerminated(proxy)
         }
       }
 

